infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftProject.java

```diff
@@ -0,0 +1,80 @@
+package org.eclipse.che.workspace.infrastructure.openshift;
+
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.openshift.api.model.Project;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+
+import java.util.Optional;
+
+/**
+ * TODO Add javadoc
+ *
+ * @author Sergii Leshchenko
+ */
+public class OpenShiftProject {
+    private final OpenShiftPods     pods;
+    private final OpenShiftServices services;
+    private final OpenShiftRoutes   routes;
+
+    public OpenShiftProject(String name,
+                            OpenShiftClientFactory clientFactory) throws InfrastructureException {
+        this.pods = new OpenShiftPods(name, clientFactory);
+        this.services = new OpenShiftServices(name, clientFactory);
+        this.routes = new OpenShiftRoutes(name, clientFactory);
+
+        try (OpenShiftClient client = clientFactory.create()) {
+            Optional<Project> projectOpt = get(name, client);
+            if (!projectOpt.isPresent()) {
+                create(name, client);
+            }
+        }
+    }
+
+    public OpenShiftPods pods() {
+        return pods;
+    }
+
+    public OpenShiftServices services() {
+        return services;
+    }
+
+    public OpenShiftRoutes routes() {
+        return routes;
+    }
+
+    public void cleanUp() throws InfrastructureException {
+        pods.delete();
+        services.delete();
+        routes.delete();
+    }
+
+    private void create(String projectName, OpenShiftClient client) throws InfrastructureException {
+        try {
+            client.projectrequests()
+                  .createNew()
+                  .withNewMetadata()
+                  .withName(projectName)
+                  .endMetadata()
+                  .done();
+        } catch (KubernetesClientException e) {
+            throw new InfrastructureException(e.getMessage(), e);
+        }
+    }
+
+    private Optional<Project> get(String projectName, OpenShiftClient client) throws InfrastructureException {
```

Comment:
Please, consider usage of null instead of  optional since it is private method and you control single place where it is used